airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -417,18 +417,23 @@ protected void assertCdcSchemaQueryable(final JsonNode config, final JdbcDatabas
   // todo: ensure this works for Azure managed SQL (since it uses different sql server agent)
   protected void assertSqlServerAgentRunning(final JdbcDatabase database) throws SQLException {
     try {
-      final List<JsonNode> queryResponse = database.queryJsons(connection -> {
-        final String sql =
-            "SELECT status_desc FROM sys.dm_server_services WHERE [servicename] LIKE 'SQL Server Agent%' OR [servicename] LIKE 'SQL Server 代理%' ";
-        final PreparedStatement ps = connection.prepareStatement(sql);
-        LOGGER.info(String.format("Checking that the SQL Server Agent is running using the query: '%s'", sql));
-        return ps;
-      }, sourceOperations::rowToJson);
-
-      if (!(queryResponse.get(0).get("status_desc").toString().contains("Running"))) {
-        throw new RuntimeException(String.format(
-            "The SQL Server Agent is not running. Current state: '%s'. Please check the documentation on ensuring SQL Server Agent is running.",
-            queryResponse.get(0).get("status_desc").toString()));
+      final Int engineEdition = database.queryInt("SELECT ServerProperty('EngineEdition')");
```

Comment:
Thank you so much for your contribution. Could you just add a comment above your change pointing to https://learn.microsoft.com/en-us/sql/t-sql/functions/serverproperty-transact-sql?view=sql-server-ver16 so that future editors can quickly see what 8 is and what other values are?

Reply:
@stephane-airbyte done!